examples/instance/terraform/plugin.go

```diff
@@ -343,6 +371,16 @@ func (p *plugin) Provision(spec instance.Spec) (*instance.ID, error) {
 
 	id := instance.ID(name)
 
+	storageType, storageName, storageProps, err := FindStorage(&tf)
```

Comment:
Actually I'd go one further....

Effectively each invocation of the `Provision` method creates a set of resources belong to the same namespace -- the `infrakit-xxxx`.  So we can even go as far as not checking for presence of a vm instance at all... and just rename every resource by qualifying the name of the resource with the generated `infrakit-xxxx`. 

So a vm instance becomes `infrakit-xxxx-<the_original_node_name_in_the_config>`.    A storage instance becomes `infrakit-xxxx-<the_original_storage_name_in_the_config>`, etc. etc. 

This will ensure we always have unique names for any invocation of the plugin for any resources. wdyt?